atmo/jobs/models.py

```diff
@@ -178,50 +208,25 @@ def is_public(self):
     def notebook_name(self):
         return self.notebook_s3_key.rsplit('/', 1)[-1]
 
+    @cached_property
+    def notebook_s3_object(self):
+        return self.provisioner.get(self.notebook_s3_key)
+
     def get_latest_run(self):
         try:
             return self.runs.latest()
         except SparkJobRun.DoesNotExist:
             return None
     latest_run = cached_property(get_latest_run, name='latest_run')
 
-    @cached_property
-    def notebook_s3_object(self):
-        return self.provisioner.get(self.notebook_s3_key)
-
-    def is_due(self, now=None):
-        """
-        Whether the scheduled Spark job is due to be run based on the
-        latest run and the configured interval in hours.
-        """
-        if now is None:
-            now = timezone.now()
-        if not self.latest_run or self.latest_run.scheduled_date is None:
-            # job has never run before
-            hours_since_last_run = float('inf')
-        else:
-            hours_since_last_run = (now - self.latest_run.scheduled_date).total_seconds() // 3600
-
-        return hours_since_last_run >= self.interval_in_hours
-
     def should_run(self):
         """Whether the scheduled Spark job should run."""
-        if not self.is_runnable:
-            return False  # the job is still running, don't start it again
-        now = timezone.now()
-        active = self.start_date <= now
-        if self.end_date is not None:
-            active = active and self.end_date >= now
-        return (
-            self.is_enabled and
-            active and
-            self.is_due(now)
-        )
+        return self.is_runnable() and self.is_enabled and self.is_due()
```

Comment:
I was a bit confused by the amount of logic behind those properties and thought methods would better match that. Also `is_enabled` is a model field, which makes it confusing if all `is_*` members are properties. Do you prefer if I were to turn `is_due` and `is_runnable` into properties again?